src/MarDyn.cpp

```diff
@@ -132,8 +132,20 @@ int run_unit_tests(const Values &options, const std::vector<std::string> &args)
  */
 int main(int argc, char** argv) {
 #ifdef ENABLE_MPI
-	MPI_Init(&argc, &argv);
-#endif
+#if defined(_OPENMP)
+	const int thread_support_level_requested = MPI_THREAD_FUNNELED; /* So far only the main thread makes MPI calls */
+#else
+	const int thread_support_level_requested = MPI_THREAD_SINGLE; /* No thread support required */
+#endif /* _OPENMP */
+	int thread_support_level_provided;
+	MPI_Init_thread(&argc, &argv, thread_support_level_requested, &thread_support_level_provided);
+	if (thread_support_level_provided < thread_support_level_requested) {
+		/* The logging infrastructure is not initalized yet, therfore perform manual output and exit. */
+		std::cerr << "ERROR: ls1 was build with OpenMP. However, the MPI implementation does not provide the necessary thread support level." << std::endl;
+		MPI_Abort(MPI_COMM_WORLD, EXIT_FAILURE);
+		::exit(EXIT_FAILURE);
```

Comment:
```suggestion
		MARDYN_EXIT("ERROR: ls1 was build with OpenMP. However, the MPI implementation does not provide the necessary thread support level.");
```

Please use the exit macro for consistent exit behavior.

Reply:
We cannot use the `mardyn_exit` macro as it relies on the logger infrastructure, which is not initialized at this point in the program (and may also dependson the MPI part). However, IMHO, we are so early in the program, that we do not need the extensive logger.

Reply:
> However, IMHO, we are so early in the program, that we do not need the extensive logger.

I respectfully disagree, especially when looking at your related PR #358. Adding exceptions to a pattern like this makes the code more confusing, prone to set bad precedent for new developers, and needs clear documentation.
I'll add a suggestion on how to tackle the issue in the mentioned PR.

Reply:
OK, then I'm looking forward to your suggestion to address the Logger issues in a cleaner way

Reply:
Thanks for the inspiration!
I will update this PR when https://github.com/ls1mardyn/ls1-mardyn/pull/359 is merged.

Reply:
Did not find time yet but will see to do so in the next weeks.